src/mod/features/small_patches.cpp

```diff
@@ -5,10 +5,38 @@
 #include "data/utils.h"
 #include "externals/FlagWork.h"
 #include "features/activated_features.h"
+#include "externals/Dpr/Battle/Logic/MainModule.h"
+#include "externals/Dpr/Battle/Logic/PokeID.h"
+
+using namespace Dpr::Battle::Logic;
 
 HOOK_DEFINE_REPLACE(FriendshipFlag) {
```

Comment:
You could also instead use a Trampoline Hook on this method to avoid having to rewrite it.
Then you'd just need to append your `&& FlagWork::GetFlag(FlagWork_Flag::FLAG_DISABLE_AFFECTION)` to the result of that